lib/cmsis_rtos_v1/cmsis_msgq.c

```diff
@@ -48,6 +48,8 @@ osStatus osMessagePut(osMessageQId queue_id, uint32_t info, uint32_t millisec)
 
 	if (retval == 0) {
 		return osOK;
+	} else if (retval == -EAGAIN) {
+		return osErrorTimeoutResource;
```

Comment:
Well.. not sure if that's correct. The spec says osErrorTimeoutResource means "no memory in the queue was available during the given time limit". Since the spec is talking about timing out only for a specific reason, i did not equate -EAGAIN with this. 
However, on a second thought, guess osErrorTimeoutResource is closer to the timeout error than osErrorResource and hence i am ok with your change.

Reply:
Okay, I did think so too, but in your osMessageGet, you have added the `osEventTimeout` and it has similar documentation:

> osEventTimeout: no message has arrived during the given timeout period

Reply:
osEventTimeout in case of osMessageGet is very straightforward since it means there was no message received in the given timeout window. However, in case of osMessagePut, it talks about "memory not being available in the queue" in that timeout window. Anyway, as i said, I am ok with your change as it seems to be closer to the actual error.

Reply:
Okay, that makes sense. Thanks Raj, addressed your review comments, and added in mailq too